### PR TITLE
Catch error on deploy for package with special comment

### PR DESCRIPTION
**Pull request by sfc-gh-pjafari**
### Pre-review checklist
   * [x] I've confirmed that instructions included in README.md are still correct after my changes in the codebase.
   * [x] I've added or updated automated unit tests to verify correctness of my new code.
   * [x] I've added or updated integration tests to verify correctness of my new code.
   * [ ] I've confirmed that my changes are working by executing CLI's commands manually on MacOS.
   * [ ] I've confirmed that my changes are working by executing CLI's commands manually on Windows.
   * [x] I've confirmed that my changes are up-to-date with the target branch.
   * [x] I've described my changes in the release notes.
   * [x] I've described my changes in the section below.

### Changes description
...
Adds a prompt for when running `app deploy` or `ws deploy` (for a package entity) that was not created by the CLI.  
This behavior already exists for version commands.

### CODE REVIEW

File: RELEASE-NOTES.md [hunk 1]
```diff
@@ -57,6 +57,7 @@
 * `snow streamlit deploy` will check for existing streamlit instance before deploying anything.
 * Fixed `snow git execute` with `/` in name of the branch.
 * `snow app` commands don't enforce ownership of the objects they manage, and rely on RBAC instead.
+* `snow app deploy` and `snow ws deploy` for package entity now allows operating on application packages created outside the CLI
```
Comment (sfc-gh-melnacouzi) [inline RELEASE-NOTES.md:60]: Maybe we should not mention ws in the release notes, as these are hidden commands.

**Review by sfc-gh-fcampbell — APPROVED** [review]

File: src/snowflake/cli/_plugins/nativeapp/commands.py [hunk 2]
```diff
@@ -452,22 +462,35 @@ def app_deploy(
...
         project_root=cli_context.project_root,
     )
     if cli_context.output_format == OutputFormat.JSON:
-        return ObjectResult(manager.get_validation_result(use_scratch_stage=True))
+        return ObjectResult(
+            manager.get_validation_result(policy=policy, use_scratch_stage=True)
+        )
 
-    manager.validate(use_scratch_stage=True)
+    manager.validate(policy=policy, use_scratch_stage=True)
```
Comment (sfc-gh-bdufour) [inline src/snowflake/cli/_plugins/nativeapp/commands.py:493]: do we need this for a scratch stage? I would have not thought so. By definition the scratch stage is something we populate, so the risk of clobbering is minimal.

File: RELEASE-NOTES.md (hunk 1 above)
Comment (sfc-gh-pjafari) [reply RELEASE-NOTES.md]: removed

File: src/snowflake/cli/_plugins/nativeapp/commands.py (hunk 2 above)
Comment (sfc-gh-pjafari) [reply src/snowflake/cli/_plugins/nativeapp/commands.py]: set policy to allow always for deploys to scratch stage

File: src/snowflake/cli/_plugins/nativeapp/commands.py [hunk 3]
```diff
@@ -452,13 +462,24 @@ def app_deploy(
 @app.command("validate", requires_connection=True)
 @with_project_definition()
 @nativeapp_definition_v2_to_v1()
-def app_validate(**options):
+def app_validate(
+    interactive: bool = InteractiveOption,
+    force: Optional[bool] = ForceOption,
```
Comment (sfc-gh-cgorrie) [inline src/snowflake/cli/_plugins/nativeapp/commands.py:467]: If the policy isn't used anywhere, we don't need the options or if tree below.

**Review by sfc-gh-fcampbell — APPROVED** [review]

**Review by sfc-gh-cgorrie — APPROVED** [review]

**Pull request merged**